src/utils.ts

```diff
@@ -55,6 +55,53 @@ export async function getIssueHtmlUrl(options: {owner: string; repo: string; iss
   }
 }
 
+/**
+ * Get list of linked issues
+ * @param {owner, repo, issue_number}  the issue options
+ * @returns {string[]}
+ */
+export async function getLinkedIssues(
+  options: {owner: string; repo: string; issue_number: number},
+  filter: {nwo: string[]}
+): Promise<string[]> {
+  try {
+    const client: github.GitHub = getAdminClient()
+
+    const linkedIssues: string[] = []
+    await client.issues
+      .listEventsForTimeline({
+        owner: options.owner,
+        repo: options.repo,
+        // eslint-disable-next-line @typescript-eslint/camelcase
+        issue_number: options.issue_number
+      })
+      .then(events => {
+        // issues is an array of all issue objects
+        core.debug(`${events}`)
+        for (const event of events.data) {
+          if (event.event === 'cross-referenced') {
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            const data: any = event
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            const item: any = data.source
+            if (item.type === 'issue' && item.issue.pull_request === undefined) {
+              const issueNWO = item.issue.repository.full_name
+              if (filter.nwo.length === 0 || filter.nwo.includes(issueNWO)) {
+                const issueURL = item.issue.html_url
+                linkedIssues.push(issueURL)
+              }
+            }
+          }
+        }
+      })
```

Comment:
I'm a bit unsure about this section. You're using `await` with the call to `client.issues.listEventsForTimeline`. You can set a variable with the events that are returned and take the code out of the `then` statement.

```suggestion
        const events = await client.issues
          .listEventsForTimeline({
            owner: options.owner,
            repo: options.repo,
            // eslint-disable-next-line @typescript-eslint/camelcase
            issue_number: options.issue_number
          })

        // issues is an array of all issue objects
        core.debug(`${events}`)
        for (const event of events.data) {
          if (event.event === 'cross-referenced') {
            // eslint-disable-next-line @typescript-eslint/no-explicit-any
            const data: any = event
            // eslint-disable-next-line @typescript-eslint/no-explicit-any
            const item: any = data.source
            if (item.type === 'issue' && item.issue.pull_request === undefined) {
              const issueNWO = item.issue.repository.full_name
              if (filter.nwo.length === 0 || filter.nwo.includes(issueNWO)) {
                const issueURL = item.issue.html_url
                linkedIssues.push(issueURL)
              }
          }
        }
      }
```

Reply:
@Chocrates can you spot check this?

Reply:
Ha, i had tried something like this as well but had run into some issues before with it. I bet I wasn't using await correctly on that round of it. Let me give this a try to see how it works locally.